tensorflow/lite/micro/kernels/cmsis-nn/fully_connected.cc

```diff
@@ -136,14 +136,16 @@ TfLiteStatus EvalQuantizedInt8(TfLiteContext* context, TfLiteNode* node,
   // The 'if' condition can be removed when null handling of bias is added to
   // arm_fully_connected_s8
   if (nullptr != GetTensorData<int32>(bias)) {
-    RuntimeShape output_shape = GetTensorShape(output);
+    const RuntimeShape output_shape = GetTensorShape(output);
     TFLITE_DCHECK_EQ(output_shape.DimensionsCount(), 2);
     const int batches = output_shape.Dims(0);
     const int output_depth = output_shape.Dims(1);
     const RuntimeShape filter_shape = GetTensorShape(filter);
     const int filter_dim_count = filter_shape.DimensionsCount();
     const int accum_depth = filter_shape.Dims(filter_dim_count - 1);
     const RuntimeShape input_shape = GetTensorShape(input);
+    TFLITE_DCHECK_EQ(input_shape.DimensionsCount(), 2);
+    TFLITE_DCHECK_EQ(batches, input_shape.Dims(0));
```

Comment:
While this is usually the case (for 2-dimensional inputs), I'm worried that for graphs without a reshape (with >2-dimensional inputs) this may not always hold true, but could still be valid. I could easily be wrong about this, but since we do not check this in the reference kernels I'm a little hesitant to add it here.

Reply:
Hi, as for `TFLITE_DCHECK_EQ(input_shape.DimensionsCount(), 2);`, I agree with you and I think that that line could be removed. However, as for the other line (`TFLITE_DCHECK_EQ(batches, input_shape.Dims(0));`) I think it is ok as it does not depend on the number of input dimensions. Are you agree with me ?

Reply:
The only issue I see is that this is not clearly specified in the reference kernel. It looks like there is an unspoken assumption that FlatSize(input) == batches * accum_depth. I do not see any specific requirements about the size of input dimension 0. I'm worried that adding a check will mean that some valid TF Micro / TF Lite models will fail unexpectedly with CMSIS-NN.

Reply:
Ok, let's skip both the checks in order to avoid that some TFLite models will fail unexpectedly when using CMSIS-NN.